src/main/java/io/supertokens/webserver/api/session/SessionAPI.java

```diff
@@ -83,4 +85,32 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
         }
     }
 
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        String sessionHandle = InputParser.getQueryParamOrThrowError(req, "sessionHandle", false);
+        assert sessionHandle != null;
+
+        try {
+            SessionInfo sessionInfo = Session.getSession(main, sessionHandle);
+
+            JsonObject result = new JsonObject();
+
+            result.addProperty("status", "OK");
+            result.add("userDataInDatabase", sessionInfo.userDataInDatabase);
+            result.add("userDataInJWT", sessionInfo.userDataInJWT);
+            result.addProperty("userId", sessionInfo.userId);
+            result.addProperty("expiry", sessionInfo.expiry);
+            result.addProperty("timeCreated", sessionInfo.timeCreated);
```

Comment:
You can use `transient` keyword in the `SessionInfo` class for properties that you don't want to be included in the JSON. With that, you can directly convert `sessionInfo` instance to JSON.